src/database/article.service.ts

```diff
@@ -32,4 +89,20 @@ export class ArticleService {
   async deleteOneById(id: Types.ObjectId) {
     return this.articleModel.deleteOne({ _id: id }).exec();
   }
-}
+
+  async findAuthorsByArticleId(articleId: string) {
+    try {
+      const article = await this.articleModel.findById(articleId).exec();
+
+      const authors = article.authors.map(author => ({
+        uid: author.uid,
+        name: author.name,
+        isAnonymous: author.isAnonymous,
+      }));
+
+      return authors;
+    } catch (error) {
+      console.log('Article Not Found')
+    }
+  }
+}
```

Comment:
We can obtain the authors using the result of `findOneById`, so you may remove this function or call `findOneById` and then return the authors. 
The parameter should align with `findOneById` (`id: Types.ObjectId`), and the function name should be `findAuthorsById`.